Separated mouse scroll and movement options (e.g. `delay` and `interval`)

Myridium opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->

Previously, scroll wheel movements and mouse cursor movements shared the same `interval` variable. This made it impossible to achieve a reasonable (e.g. not blazingly fast) scrolling speed while also having the cursor speed configured to be reasonable.

I have implemented separate timers for the scroll wheel and mouse cursor movements, and added appropriate options. This makes breaking changes in order to keep a consistent naming convention (e.g. `MOUSEKEY_DELAY` has been changed to `MOUSEKEY_XY_DELAY`).

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [x] Bugfix
- [x] Enhancement/optimization

## Issues Fixed or Closed by This PR

* #1434
* #5690
* #3370

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

noroadsleft commented:
I haven't looked over the code yet at all, but based on the description and docs, it seems this may overlap with or supersede #6685.

Myridium commented:
Thanks for the response @noroadsleft . There was a bug I fixed in my commits which I think could have been present in the existing code: the key repeat counter was not being reset once a key was released. This made the delay work _inconsistently_ and in a reproducible way. I do not understand how it was working. 

I cloned #6685 into a fresh directory and am not able to compile. `keymap_common.h` does not exist but is imported by some other file. I'm not sure if I'm doing something wrong or if #6685  is broken somehow.

Myridium commented:
Hello, can someone please tell me how to change the branch on my own fork which this pull request references? I would like to continue development on the `master` branch of my fork and I thought this pull request would be merged by now.

drashna commented:
There isn't a way to retarget without closing the PR, pushing the changes to a new branch, and opening a new PR. Which is fine.

Myridium commented:
I see this change was to be merged in Q1 then this was revoked unexpectedly. What is going on? Will this be merged?

noroadsleft commented:
Hm, I don't quite recall why I unlabeled this for the Q1 merge, though I note it has yet to be reviewed.

Anyway, as it stands now, this *can't* be merged to `master` because of merge conflicts to:

- docs/feature_mouse_keys.md
- tmk_core/common/mousekey.c
- tmk_core/common/mousekey.h

... probably from 6685.

The merge conflicts need to be fixed, and we need some QMK members to review this PR.

I'll cycle back around later today and see if I can get to it, but I neither use mouse keys nor am skilled at C, so this code may be over my head.

noroadsleft commented:
Okay, I've read this over. I get the gist of what it's doing, but not the specifics.

Also, does this code do anything that PR6685 didn't do? I can't quite tell yet.

Myridium commented:
Looks like PR6685 captures the core functionality. However my PR did improve the documentation also, and cleaned up the existing variables and in-code documentation a fair bit, also improving naming consistency of existing constants. I also added the appropriate entries to the debugging tools in e.g. tmk_core/common/command.c which PR6685 did not do. For consistency, these should have been added.

For these reasons, I think it was a mistake to merge #6685 instead of this pull request. Oh well.

tzarc commented:
> 
> For these reasons, I think it was a mistake to merge #6685 instead of this pull request. Oh well.

It happens. That doesn't stop iterating over the problem, so if what you've proposed is better or more comprehensive, we'd be happy to get it in to improve matters.

I'm going to have to retarget this to the `develop` branch as it's got core changes, though.

Myridium commented:
I should note that I've since discovered an issue with this PR. Moving the mouse temporarily increases the scroll speed dramatically. So, if you're scrolling and move the mouse at the same time, the scrolling goes very rapidly until you stop moving the mouse.
